### PR TITLE
Add documentation.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/app.dart [hunk 1]
```diff
@@ -443,6 +443,15 @@ class DevToolsScreen<C extends Object?> {
   ///
   /// If [createController] and [controller] are both null, [screen] will be
   /// responsible for creating and maintaining its own controller.
+  ///
+  /// The controller should expect to be already connected to an application
+  /// at the time of creation.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:448]: This is not true. A controller does not require a connected application.

File: packages/devtools_app/lib/src/app.dart [hunk 2]
```diff
@@ -443,6 +443,15 @@ class DevToolsScreen<C extends Object?> {
...
+  /// But it should invoke unawaitable `serviceConnection.serviceManager.onServiceAvailable`
+  /// in the constructor to make sure the service is ready to be used.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:450]: In the controller initialization, if logic requires a connected [VmService] object (`serviceConnection.serviceManager.service`), then the controller should first await
the `serviceConnection.serviceManager.onServiceAvailable` future to ensure the service has
been initialized.

File: packages/devtools_app/lib/src/app.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/app.dart:448]: rephrased

File: packages/devtools_app/lib/src/app.dart [hunk 3]
```diff
@@ -444,11 +444,10 @@ class DevToolsScreen<C extends Object?> {
...
   /// responsible for creating and maintaining its own controller.
   ///
-  /// The controller should expect to be already connected to an application
-  /// at the time of creation.
-  /// But it should invoke unawaitable `serviceConnection.serviceManager.onServiceAvailable`
-  /// in the constructor to make sure the service is ready to be used.
-  ///
+  /// In the controller initialization, if logic requires a connected [VmService]
+  /// object (serviceConnection.serviceManager.service), then the controller should first await
+  /// the serviceConnection.serviceManager.onServiceAvailable future to ensure the service has
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:449]: back tics for the code referenced here

**Review by kenzieschmoll — APPROVED** [review]
lgtm w/ one nit

**Pull request merged**